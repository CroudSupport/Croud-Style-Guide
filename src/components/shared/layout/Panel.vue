<template>
    <div class="ui segments">
        <div class="ui basic segment text menu">
            <div class="item">
                <h2 class="heading ui-sortable-handle">{{ title }}</h2>
            </div>
            <div class="right item">
                <slot name="extra"></slot>
                <i class="circular angle icon" :class="{down: !collapse, left: collapse }" @click="collapse = !collapse"></i>
            </div>
        </div>
        <div class="ui basic">
            <transition name="slide-down">
                <div v-show="!collapse">
                    <slot></slot>
                </div>
            </transition>
        </div>
    </div>
</template>

<script>
    /**
     * Generic Croud card component
     *
     * @example ./croud-panel.md
     */
    export default {
        name: 'croud-panel',

        props: {
            /**
             * Set the large title in the top left of the card
             */
            title: {
                type: String,
                default: 'Title',
            },
        },

        data() {
            return {
                collapse: false,
            }
        },
    }
</script>

<style lang="scss" scoped>
    @import '../../../resources/sass/variables/_all.scss'; 

    .ui.black.inverted.segment {
        background-color: $croud-colour-header !important;
    }

    .ui.basic {
        position: relative;
    }

    .slide-down-enter-active, .slide-down-leave-active {
        max-height: 1000px;
        transition: all .3s ease;
    }

    .slide-down-enter, .slide-down-leave-to {
        max-height: 0 !important;
        overflow-y: hidden;
        opacity: 0;
    }

    h2 {
        font-family: $croud-font-headline;
        border-bottom: 3px solid #f4c91b;
        text-transform: uppercase;
        font-weight: 400;
        font-size: 26px;
        padding-bottom: 0.1em;
        margin-bottom: 0.8em;
        margin-top: 0.4em;
    }

    .ui.text.menu {
        padding-top: 0;
        padding-bottom: 0;
    }

    .ui.segments {
        background: #fff;
    }

    .ui.basic {
        /*padding: 0 1em;*/
    }
</style>

<style lang="scss">

    .ui.segment.filterbox.text.menu {
        /*margin-left: -1em
        margin-right: -1em*/
        border-bottom: 1px solid hsla(0,0%,39%,.07);

        + .loader-container {
            margin-top: 0;
            
            .ui.table {
                margin-bottom: 0;
            }
        }
    }
</style>
