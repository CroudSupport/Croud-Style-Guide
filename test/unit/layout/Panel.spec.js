import Vue from 'vue'
import Panel from '../../../src/components/shared/layout/Panel'

const Constructor = Vue.extend(Panel)

// const vm = new Constructor().$mount()


const vm = new Constructor({
    propsData: {
        title: 'Test Panel',
        showCollapseToggle: true,
    },
}).$mount()

describe('Panel', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    it('should show the show / collapse button', () => {
        expect(vm.$el.querySelectorAll('.circular.angle.icon').length).toBe(1)
    })

    it('should show the text segment', () => {
        expect(vm.$el.querySelectorAll('.text.menu').length).toBe(1)
    })
})

describe('showCollapseToggle is false', () => {
    beforeEach(() => {
        vm.showCollapseToggle = false
    })

    it('shouldnt show the show / collapse button', () => {
        expect(vm.$el.querySelectorAll('.circular.angle.icon').length).toBe(0)
    })
})

describe('No title length and showCollapseToggle is false', () => {
    beforeEach(() => {
        vm.title = ''
        vm.showCollapseToggle = false
    })

    it('shouldnt show the text segment', () => {
        expect(vm.$el.querySelectorAll('.text.menu').length).toBe(0)
    })
})
