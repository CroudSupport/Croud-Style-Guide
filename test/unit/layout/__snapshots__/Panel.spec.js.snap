// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should match the snapshot 1`] = `
<div
  class="ui segments"
>
  <div
    class="ui basic segment text menu"
  >
    <div
      class="item"
    >
      <h2
        class="heading ui-sortable-handle"
      >
        Test Panel
      </h2>
    </div>
    <div
      class="right item"
    >
      <i
        class="chevron icon down"
      />
    </div>
  </div>
  <div
    class="ui basic"
  >
    <div />
  </div>
</div>
`;
